src/snowflake/snowpark/dataframe.py

```diff
@@ -2060,6 +2060,21 @@ def join(
                 using_columns = [using_columns]
             elif isinstance(using_columns, Column):
                 using_columns = using_columns
+            elif (
+                isinstance(using_columns, list)
+                and len(using_columns) > 0
+                and not all([type(col) == str for col in using_columns])
+            ):
+                bad_col, bad_idx = None, 0
```

Comment:
NIT: `bad_col, bad_idx = next(col, idx for idx, col in enumerate(using_columns) if type(col) !=str)`

Reply:
that's cool